File: app/src/main/java/msc/app/embalsespuertorico/MainActivityPager.java
package msc.app.embalsespuertorico;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

/**
 * Created by Moises Cardona on 2/8/2017.
 */

public class MainActivityPager extends FragmentStatePagerAdapter {
    private int tabCount;
    public MainActivityPager(FragmentManager fm, int tabCount) {
        super(fm);
        this.tabCount= tabCount;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new MainActivity();
            case 1:
                return new LookInMap();
            default:
                return null;
        }
    }
    @Override
    public int getItemPosition(@NonNull Object object) {
        return POSITION_NONE;
    }
    @Override
    public int getCount() {
        return tabCount;
    }
}

File: app/src/main/java/msc/app/embalsespuertorico/MainActivityPager.kt
package msc.app.embalsespuertorico

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.PagerAdapter

/**
 * Created by Moises Cardona on 2/8/2017.
 */

class MainActivityPager(fm: FragmentManager, private val tabCount: Int) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> MainActivity()
            1 -> LookInMap()
            else -> MainActivity()
        }
    }

    override fun getItemPosition(`object`: Any): Int {
        return PagerAdapter.POSITION_NONE
    }

    override fun getCount(): Int {
        return tabCount
    }
}
